contractcourt/chain_arbitrator.go

```diff
@@ -1057,13 +1057,77 @@ type forceCloseReq struct {
 	closeTx chan *wire.MsgTx
 }
 
+// ForceCloseContractOption is used for providing functional options to callers
+// of ForceCloseContract().
+type ForceCloseContractOption func(*ChannelArbitrator)
```

Comment:
Okay could you please explain why this is so? I got that it is "too powerful" but that was not backed up by anything really.

Reply:
It is too powerful because you are using a full continuation here. Someone could supply a function that does anything and it is delegating the process of calling the "log" functions to the argument. If you look at ChainArbitrator.ForceCloseContract it is simply calling out to the supplied `opt` do do the log action. However, all of what you need to do with this is finitely enumerable and we have no reason to believe it ever won't be. So you should just take a regular data parameter and then call the appropriate log action depending on that value.

Continuation passing has its uses but I don't think this is a good instance of it.

That said, I wouldn't waste time taking a look at these smaller comments before zooming out and reworking the approach. See my [other comment](https://github.com/lightningnetwork/lnd/pull/8072#issuecomment-1841965713). I think @ziggie1984 has the right idea of what to do here.

Reply:
I don't quite get this @yyforyongyu. Care explaining more about this?